var/spack/repos/builtin/packages/charm/package.py

```diff
@@ -92,6 +92,24 @@ class Charm(Package):
     variant("production", default=True, description="Build charm++ with all optimizations")
     variant("tracing", default=False, description="Enable tracing modules")
 
+    # FIXME: backend=mpi also provides mpi, but this can't be expressed in
+    # spack currently
+    # FIXME: shorten this to 'build-target!=charm++'
```

Comment:
You could shorten it quite a bit with a for-loop.

Reply:
Done, thanks! Do you have a suggestion on how to fix the `backend=mpi` situation? Currently, this results in an error message `==> Error: maximum recursion depth exceeded`, since in this case we have `depends_on('mpi')` as well as `provides('mpi')`.

Reply:
No idea.

Reply:
Ok. Not having this for the mpi backend might be enough for now I think.